gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1276 @@
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named '%s' does not exist", zoneName));
+    }
+    // take a sorted snapshot of the current change list
+    NavigableMap<Integer, Change> changes = new TreeMap<>();
+    for (Change c : zoneContainer.changes()) {
+      if (c.getId() != null) {
+        changes.put(Integer.valueOf(c.getId()), c);
+      }
+    }
+    String[] fields = (String[]) options.get("fields");
+    String sortOrder = (String) options.get("sortOrder");
+    String pageToken = (String) options.get("pageToken");
+    Integer maxResults = options.get("maxResults") == null
+        ? null : Integer.valueOf((String) options.get("maxResults"));
+    // we are not reading sortBy as the only key is the change sequence
+    NavigableSet<Integer> keys;
+    if ("descending".equals(sortOrder)) {
+      keys = changes.descendingKeySet();
+    } else {
+      keys = changes.navigableKeySet();
+    }
+    Integer from = null;
+    try {
+      from = Integer.valueOf(pageToken);
+    } catch (NumberFormatException ex) {
+      // ignore page token
+    }
+    keys = from != null && keys.contains(from) ? keys.tailSet(from, false) : keys;
+    NavigableMap<Integer, Change> fragment =
+        from != null && changes.containsKey(from) ? changes.tailMap(from, false) : changes;
+    boolean sizeReached = false; // maximum result size was reached, we should not return more
+    boolean hasMorePages = false; // should next page token be included in the response?
+    LinkedList<String> serializedResults = new LinkedList<>();
+    String lastChangeId = null;
+    for (Integer key : keys) {
+      Change change = fragment.get(key);
+      if (sizeReached) {
+        // we do not add this, just note that there would be more and there should be a token
+        hasMorePages = true;
+        break;
+      } else {
+        lastChangeId = change.getId();
+        try {
+          serializedResults.addLast(jsonFactory.toString(
+              OptionParsers.extractFields(change, fields)));
+        } catch (IOException e) {
+          return Error.INTERNAL_ERROR.response(String.format(
+              "Error when serializing change %s in managed zone %s in project %s",
+              change.getId(), zoneName, projectId));
+        }
+      }
+      sizeReached = (maxResults != null) && maxResults.equals(serializedResults.size());
+    }
+    boolean includePageToken =
+        hasMorePages && (fields == null || ImmutableList.copyOf(fields).contains("nextPageToken"));
+    return toListResponse(serializedResults, "changes", lastChangeId, includePageToken);
+  }
+
+  /**
+   * Validates a zone to be created.
+   */
+  private static Response checkZone(ManagedZone zone) {
+    if (zone.getName() == null) {
+      return Error.REQUIRED.response(
+          "The 'entity.managedZone.name' parameter is required but was missing.");
+    }
+    if (zone.getDnsName() == null) {
```

Comment:
Tangential to this PR: in the `Zone` and `ZoneInfo` classes, it seems like this field isn't required to create a builder or build the object.  Should `dnsName` be made a mandatory field in those classes?